test/parallel/test-crypto-dh-stateless.js

```diff
@@ -196,13 +196,15 @@ for (const [params1, params2] of [
 test(crypto.generateKeyPairSync('ec', { namedCurve: 'secp256k1' }),
      crypto.generateKeyPairSync('ec', { namedCurve: 'secp256k1' }));
 
-assert.throws(() => {
-  test(crypto.generateKeyPairSync('ec', { namedCurve: 'secp256k1' }),
-       crypto.generateKeyPairSync('ec', { namedCurve: 'secp224k1' }));
-}, {
-  name: 'Error',
-  code: 'ERR_OSSL_EVP_DIFFERENT_PARAMETERS'
-});
+if (!process.config.variables.node_shared_openssl) {
+  assert.throws(() => {
+    test(crypto.generateKeyPairSync('ec', { namedCurve: 'secp256k1' }),
+         crypto.generateKeyPairSync('ec', { namedCurve: 'secp224k1' }));
+  }, {
+    name: 'Error',
+    code: 'ERR_OSSL_EVP_DIFFERENT_PARAMETERS'
+  });
+}
```

Comment:
IMO, it's better to feature-detect it. Being linked against a shared copy of openssl doesn't have to mean the curve is unsupported.
```js
{
  let secp224k1;
  try {
    secp224k1 = crypto.generateKeyPairSync('ec', { namedCurve: 'secp224k1' });
  } catch (e) {
    if (e.message !== 'Invalid ECDH curve name') throw e;
  }
  if (secp224k1) {
    // ...
  }
}
```

Reply:
> Being linked against a shared copy of openssl doesn't have to mean the curve is unsupported.

My reasoning was that we know that the statically linked OpenSSL has this curve, but I'm happy to go with feature-detection like you suggested. Thanks.

Reply:
Your proposed fix was fine in isolation but such patterns tend to propagate across the test suite (because people like to crib existing code), to contexts where checking `node_shared_openssl` won't be appropriate. Best to set a good example. :-)